home-service/src/main/java/com/softserveinc/ita/homeproject/homeservice/service/impl/UserServiceImpl.java

```diff
@@ -29,42 +33,74 @@
 
     private static final String USER_NOT_FOUND_FORMAT = "User with id: %d is not found";
 
+    private static final String EMAILS_NOT_MATCH = "The e-mail to which the token was sent "
+            + "does not match provided";
+
     private final UserRepository userRepository;
 
     private final RoleRepository roleRepository;
 
+    private final UserCooperationRepository userCooperationRepository;
+
+    private final InvitationRepository invitationRepository;
+
+    private final ApartmentInvitationServiceImpl apartmentInvitationService;
+
+    private final CooperationInvitationService cooperationInvitationService;
+
     private final PasswordEncoder passwordEncoder;
 
     private final ServiceMapper mapper;
 
     @Override
     @Transactional
     public UserDto createUser(UserDto createUserDto) {
+
+        User toCreate = mapper.convert(createUserDto, User.class);
+
         if (userRepository.findByEmail(createUserDto.getEmail()).isEmpty()) {
-            User toCreate = mapper.convert(createUserDto, User.class);
             toCreate.setPassword(passwordEncoder.encode(createUserDto.getPassword()));
             toCreate.setEnabled(true);
             toCreate.setExpired(false);
-            toCreate.setRoles(Set.of(roleRepository.findByName(USER_ROLE)
-                    .orElseThrow(() -> new NotFoundHomeException("RoleDto not found."))));
             toCreate.setCreateDate(LocalDateTime.now());
             toCreate.getContacts().forEach(contact -> {
                 contact.setUser(toCreate);
                 contact.setEnabled(true);
             });
 
             userRepository.save(toCreate);
-
+            checkAndSaveUserByInvitation(createUserDto);
             return mapper.convert(toCreate, UserDto.class);
         }
         throw new AlreadyExistHomeException("User with email " + createUserDto.getEmail() + " is already exists");
     }
 
+    private void checkAndSaveUserByInvitation(UserDto userDto) {
+
+        var invitation = invitationRepository.findInvitationByRegistrationToken(userDto.getRegistrationToken())
+                .filter(invitation1 -> invitation1.getStatus().equals(InvitationStatus.PROCESSING))
+                .orElseThrow(() -> new NotFoundHomeException("Invitation with provided token not found"));
+
+        validateEmailsMatching(invitation.getEmail(), userDto.getEmail());
+
+        if (invitation.getType().equals(InvitationType.APARTMENT)) {
+            apartmentInvitationService.acceptUserInvitation(invitation);
+        } else {
```

Comment:
Please add check for another invitation status. If we may add another one in future, this is a potential place for bug.

Reply:
Thanks for notice. Added switch statement